### PR TITLE
[Typescript Angular] Set ResponseType for HttpClient Call

### LINKED ISSUE #2035: [BUG] [typescript-angular] Missing handling of text responses
##### Description

When the returned content type of an API is a text type (e.g. `text/plain`), the Angular generated service doesn't set the `responseType: text` option (as shown in Angular documentation: https://angular.io/guide/http#requesting-non-json-data)

##### openapi-generator version
3.3.4

##### OpenAPI declaration file content or url
```yaml
openapi: 3.0.0
info:
  title: MyAPI
  description: MY API
  version: 0.0.0
servers:
  - url: http://localhost:8080/api/
paths:
  /example:
    get:
      operationId: getExample
      responses:
        "200":
          description: "Success"
          content:
            text/plain:
              schema:
                type: string
```

##### Command line used for generation
`docker run --rm -v ${PWD}:/local openapitools/openapi-generator-cli:v3.3.4 generate -i /local/swagger.yaml  -g typescript-angular\ -o /local/src/app/api\ -DngVersion=6.0.0

##### Suggest a fix
This is the generated service:
```typescript
    public getExample(observe?: 'body', reportProgress?: boolean): Observable<string>;
    public getExample(observe?: 'response', reportProgress?: boolean): Observable<HttpResponse<string>>;
    public getExample(observe?: 'events', reportProgress?: boolean): Observable<HttpEvent<string>>;
    public getExample(observe: any = 'body', reportProgress: boolean = false ): Observable<any> {

        let headers = this.defaultHeaders;

        // to determine the Accept header
        let httpHeaderAccepts: string[] = [
            'text/plain'
        ];
        const httpHeaderAcceptSelected: string | undefined = this.configuration.selectHeaderAccept(httpHeaderAccepts);
        if (httpHeaderAcceptSelected !== undefined) {
            headers = headers.set('Accept', httpHeaderAcceptSelected);
        }

        // to determine the Content-Type header
        const consumes: string[] = [
        ];

        return this.httpClient.get<string>(`${this.configuration.basePath}/example`,
            {
                withCredentials: this.configuration.withCredentials,
                headers: headers,
                observe: observe,
                reportProgress: reportProgress
            }
        );
    }
```

Instead the generator should call the get method with this format:
```typescript
        return this.httpClient.get(`${this.configuration.basePath}/example`,
            {
                withCredentials: this.configuration.withCredentials,
                headers: headers,
                observe: observe,
                reportProgress: reportProgress,
                responseType: 'text'
            }
        );
``` 

Without the setting of this option. the http client will return an error instead of the text content of the called API

### PR DESCRIPTION
**Pull request by karismann**
### PR checklist

- [x] Read the [contribution guidelines](https://github.com/openapitools/openapi-generator/blob/master/CONTRIBUTING.md).
- [x] Ran the shell script under `./bin/` to update Petstore sample so that CIs can verify the change. (For instance, only need to run `./bin/{LANG}-petstore.sh`, `./bin/openapi3/{LANG}-petstore.sh` if updating the {LANG} (e.g. php, ruby, python, etc) code generator or {LANG} client's mustache templates). Windows batch files can be found in `.\bin\windows\`. If contributing template-only or documentation-only changes which will change sample output, be sure to [build the project](https://github.com/OpenAPITools/openapi-generator#14---build-projects) first.
- [x] Filed the PR against the [correct branch](https://github.com/OpenAPITools/openapi-generator/wiki/Git-Branches): `master`, `4.1.x`, `5.0.x`. Default: `master`.
- [x] Copied the [technical committee](https://github.com/openapitools/openapi-generator/#62---openapi-generator-technical-committee) to review the pull request if your PR is targeting a particular programming language.

@TiFu (2017/07) @taxpon (2017/07) @sebastianhaas (2017/07) @kenisteward (2017/07) @Vrolijkx (2017/09) @macjohnny (2018/01) @nicokoenig (2018/09) @topce (2018/10)

### Description of the PR

- fix #2035 :
fit HttpClient response type depend of the accept response send in the request  : 
default => json
application/octet-stream => blob
text/* => text
- minor whitespace indentation removing

### CODE REVIEW

File: modules/openapi-generator/src/main/resources/typescript-angular/api.service.mustache [hunk 1]
```diff
@@ -319,21 +319,24 @@ export class {{classname}} {
...
+            {{/hasQueryParams}}
+            withCredentials: this.configuration.withCredentials,
+            headers: headers,
+            observe: observe,
+            reportProgress: reportProgress,
+            {{^isResponseFile}}
+            responseType: this.configuration.selectResponseType(httpHeaderAcceptSelected)
+            {{/isResponseFile}}
+            {{#isResponseFile}}
+            responseType: 'blob',
```
Comment (macjohnny) [inline modules/openapi-generator/src/main/resources/typescript-angular/api.service.mustache]:
```suggestion
            responseType: 'blob' as 'blob',
```

File: modules/openapi-generator/src/main/resources/typescript-angular/api.service.mustache [hunk 2]
```diff
@@ -319,21 +319,24 @@ export class {{classname}} {
...
+        const httpOptions: Object = {
+            {{#hasQueryParams}}
+            params: queryParameters,
+            {{/hasQueryParams}}
+            withCredentials: this.configuration.withCredentials,
+            headers: headers,
+            observe: observe,
+            reportProgress: reportProgress,
+            {{^isResponseFile}}
+            responseType: this.configuration.selectResponseType(httpHeaderAcceptSelected)
```
Comment (macjohnny) [inline modules/openapi-generator/src/main/resources/typescript-angular/api.service.mustache]: I think the type needs to be set explicitly, can we determine the response type when generating the api?

Comment (karismann) [reply modules/openapi-generator/src/main/resources/typescript-angular/api.service.mustache]: yes you're right , this change doesn't work

**Comment by FallenRiteMonk** [comment]
@karismann why did you close this PR?

**Pull request closed without merging**